umbral-pre/src/hashing.rs

```diff
@@ -56,27 +52,40 @@ pub fn unsafe_hash_to_point(dst: &[u8], data: &[u8]) -> Option<CurvePoint> {
     None
 }
 
-pub(crate) struct ScalarDigest(Sha256);
+// Wraps Sha256 for easier replacement, and standardizes the use of DST.
+struct Hash(Sha256);
 
-impl ScalarDigest {
-    pub fn new() -> Self {
-        Self(Sha256::new())
+// Can't be put in the `impl` in the current version of Rust.
+pub type HashOutputSize = <Sha256 as Digest>::OutputSize;
+
+impl Hash {
+    pub fn new_with_dst(dst: &[u8]) -> Self {
+        let dst_len = (dst.len() as u32).to_be_bytes();
+        Self(Sha256::new()).chain_bytes(dst_len).chain_bytes(dst)
     }
 
-    pub fn new_with_dst(bytes: &[u8]) -> Self {
-        Self::new().chain_bytes(bytes)
+    pub fn chain_bytes<T: AsRef<[u8]>>(self, bytes: T) -> Self {
+        Self(self.0.chain(bytes.as_ref()))
+    }
+
+    pub fn digest(self) -> Sha256 {
+        self.0
```

Comment:
What does `self.0` mean here? I find it a bit obscure

Reply:
That's how you address the fields of a tuple-like type. Usually if you just want to wrap a single type (as is done here for `Sha256`), you don't create a structure with a named field, but instead define it as a tuple, that is
```
struct Hash(Sha256);
```
instead of 
```
struct Hash {
    digest: Sha256
}
```
And you address this field as `self.0` instead of `self.digest`. 

There's no strict rule on where to use these anonymous fields, but generally if there's only one field, and the wrapper is pretty thin, the convention is to do that.